protocol-spec/generic.md

```diff
@@ -134,5 +142,197 @@
 ]
 ```
 
+## VibrateCmd
+
+**Description:** Causes a toy that supports vibration to run specific vibration motors at a certain speeds. In order to abstract the dynamic range of different toys, the values are sent as a dictionary of vibration motor indexes against floats with a range of \[0.0-1.0\]
+
+**Introduced In Version:** 1
+
+**Message Version:** 1
+
+**Fields:**
+
+* _Id_ \(unsigned int\): Message Id
+* _DeviceIndex_ \(unsigned int\): Index of device
+* _Speeds_ \(array\): Vibration speeds
+  * _Index_ \(unsigned int\): Index of vibration motor
```

Comment:
Might it just be worth having this as an array of (number, number) representing (index, speed)? I realize that's a little more terse than the rest of what we've been doing here, so if this is just following our format otherwise, I guess that works?

Reply:
I guess we could, but this is more readable and was really easy for the JSON object mapping to cope with. For other commands with more complex requirements, readability might be more important and keeping a consistent layout seemed important to me.

Reply:
Ok that's fine then.